README.md

```diff
@@ -159,6 +159,207 @@ python ./scripts/train_hf_ner.py ./data/annotations/train.spacy ./data/annotatio
+    --verbose \
+    --exclusive-classes \
+    -F recipes/openai_textcat.py
+```
+
+<img src="https://user-images.githubusercontent.com/12949683/214230269-bdacbbc8-8edc-4be5-8334-0b7eaf4712e0.png" width="600" />
+
+#### Writing templates
+
+We write these prompts as a .jinja2 template that can also take in examples for
+few-shot learning. You can create your own [template](templates) and provide it
+to the recipe with the `--prompt-path` or `-p` option.  Additionally, with
+`--examples-path` or `-e` you can set the file path of a .y(a)ml or .json file
+that contains additional examples:
+
+```bash
+python -m prodigy textcat.openai.correct my_textcat_data \
+    ./data/reddit_r_cooking_sample.jsonl \
+    --labels recipe \
+    --prompt-path ./templates/textcat_prompt.jinja2 \
+    --examples-path ./examples/textcat_binary.yaml -n 2 \
+    -F ./recipes/openai_textcat.py
+```
+
+Similar to the NER recipe, this recipe also converts the predictions into an 
+annotation task that can be rendered with Prodigy. For binary classification, we
+use the [`classification`](https://prodi.gy/docs/api-interfaces#classification)
+interface with custom HTML elements, while for multilabel or multiclass text
+categorization, we use the
+[`choice`](https://prodi.gy/docs/api-interfaces#choice) annotation interface.
+Notice that we include the original prompt and ChatGPT's response in the UI. 
+
+Lastly, you can use the `--verbose` or `-v` flag to show the exact prompt and
+response on the terminal. Note that because the requests to the API are batched,
+you might have to scroll back a bit to find the current prompt.
+
+## `textcat.openai.fetch`: Fetch text categorization examples up-front
+
+The `textcat.openai.fetch` recipe allows us to fetch a large batch of examples
+upfront. This is helpful when you are with a highly-imbalanced data and interested
+only in rare examples.
+
+```bash
+python -m prodigy textcat.openai.fetch input_data.jsonl predictions.jsonl \
+    --labels Recipe \
+    -F ./recipes/openai_textcat.py
+```
+
+This will create a `predictions.jsonl` file that can be loaded with the
+[`textcat.manual`](https://prodi.gy/docs/recipes#textcat-manual) recipe.
+
+Note that the OpenAI API might return "429 Too Many Request" errors when
+requesting too much data at once - in this case it's best to ensure you only
+request 100 or so examples at a time and take a look at the [API's rate
+limits](https://help.openai.com/en/articles/5955598-is-api-usage-subject-to-any-rate-limits).
+
+### Working with imbalanced data
+
+The `textcat.openai.fetch` recipe is suitable for working with datasets where
+there is severe class imbalance. Usually, you'd want to find examples of the
+rare class rather than annotating a random sample. From there, you want to
+upsample them to train a decent model and so on.
+
+As it turns out, **large language models (LLMs) are effective at finding the
+rare class.** Using the [Reddit r/cooking dataset](data), we prompted ChatGPT to
```

Comment:
Given the amount of effort it can take to get OpenAI to do that you want, it might better to tone down this claim a bit. 

Maybe this? 

```suggestion
This is where large language models like OpenAI might help.
Using the [Reddit r/cooking dataset](data), we prompted ChatGPT to
```